lib/doctest.js

```diff
@@ -500,73 +555,169 @@ function rewrite$coffee(options, input) {
   );
 }
 
-function functionEval(source) {
+function functionEval(options, source, logFunction) {
   //  Functions created via the Function function are always run in the
   //  global context, which ensures that doctests can't access variables
   //  in _this_ context.
   //
-  //  The `evaluate` function takes one argument, named `__doctest`.
-  var evaluate = Function ('__doctest', source);
+  //  The `evaluate` function takes one or two arguments, `__doctest` is
+  //  the doctest runtime broker, and optionally it takes a logger function.
```

Comment:
Could we require the logging function here too?

Reply:
The user decides the name for the logging function. I'm not sure if we want to inject a function with some default name, so I made the whole thing optional.